src/commands/executeFunction.ts

```diff
@@ -48,6 +48,8 @@ export async function executeFunction(context: IActionContext, node?: FunctionTr
         const data: string = await context.ui.showInputBox({ prompt, value, stepName: 'requestBody' });
         try {
             functionInput = <{}>JSON.parse(data);
+            // stringify JSON object to match the format in the portal
+            functionInput = <{}>JSON.stringify(functionInput, undefined, 2);
```

Comment:
The whole point of the try/catch is to `JSON.parse` the data, so if you're just going to re-stringify it then you could get rid of the whole try/catch.

More importantly though, I believe http triggers need this to be JSON.parsed and that's probably why I added this try/catch in the first place. The default request body is `{ "name": "Azure" }` and it should return "Hello, Azure! ..."

Reply:
Ftr, Nathan's gif is why I looked at this PR 👀

Reply:
We were experiencing issues with users of the service bus triggers. The service bus queue triggers weren't accepting JSON objects.  Apparently the portal takes the object and stringifies it instead.

That being said, Eric has a point about the HTTP trigger expecting a JSON object and that working. So all of this seems pretty opaque 😐

Reply:
The default request body is empty in the portal for an http trigger, but if I put `{ "name": "Azure" }` then it correctly responds with `Hello, Azure...`. So there's obviously something else the portal is doing here that VS Code isn't

Also ftr, I would prioritize maintaining http support over fixing a long-standing service bus bug

Reply:
I think we could either:

- Check to see if it's an HTTP trigger and skip this behavior
- Try the original method first-- if there's a 400 error, try stringifying it

Reply:
I think I may prefer checking for an HTTP trigger since that would be simpler code, but I could see arguments for both.

Reply:
Though one drawback is that we may be breaking more triggers than just HTTP.